### PR TITLE
chore: v0.9.38

### PR DESCRIPTION
**Pull request by enthusiastmartin**
based on #174.

Notable changes:

- required to install protobuf compiler
  sudo apt-get install protobuf-compiler
- for math repo, required to install m4
  sudo apt-get install m4
- pallet stuff renamed
  - Event → RuntimeEvent
  - From<Event> + IsType<::RuntimeEvent>;
  - Origin → RuntimeOrigin
  - Call → RuntimeCall
- extrinsics need call_index ( otherwise depreated warning )
  #[pallet::call_index(0)]
- orml traits
  OnDust trait move into submodule called currency
- NFT pallet
  InspectEnumerable trait changed - needs closer look to verify f reimplemented correcy.
- pallet currencies
  check if changes in orml currencies to ensure no bugs fixed ?
- orml tokens
  callbacks OnDust,OnNewAccoutn and OnKilledAccount combined into one CurrencyHooks.

### CODE REVIEW

**Comment by apopiak** [comment]
we should make sure to include https://github.com/galacticcouncil/warehouse/pull/206 (just merged)

File: test-utils/src/rpc_service.rs
Comment (mrq1911) [inline test-utils/src/rpc_service.rs:1]: do we need this file?

File: transaction-pause/src/tests.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Originally created by Acala. Modified by GalacticCouncil.
+// RuntimeOriginally created by Acala. Modified by GalacticCouncil.
```
Comment (mrq1911) [inline transaction-pause/src/tests.rs:1]: hh

File: duster/src/migration.rs [hunk 2]
```diff
@@ -19,7 +19,6 @@ use crate::Config;
 use frame_support::weights::Weight;
 use sp_std::vec::Vec;
 
-/// Storage names are changed from Classes to Collections and from Instances to Items.
 pub mod v1 {
```
[the remark this answers is not in the document]
Comment (Roznovjak) [reply duster/src/migration.rs:22]: Yes, based on the storage version in Basilisk and Hydra, we will need it.

File: nft/src/lib.rs [hunk 3]
```diff
@@ -233,6 +237,7 @@ pub mod pallet {
         /// Parameters:
         /// - `origin`: The collection owner.
         /// - `collection_id`: The identifier of the asset collection to be destroyed.
+        #[pallet::call_index(4)]
```
[the remark this answers is not in the document]
Comment (Roznovjak) [reply nft/src/lib.rs:240]: No, but based on my assumption and [this](https://github.com/paritytech/substrate/pull/12891#pullrequestreview-1213633451) comment indexes are assigned sequentially and in order.

File: nft/src/tests.rs [hunk 4]
```diff
@@ -75,7 +75,7 @@ fn create_collection_works() {
         // reserved collection ID
         assert_noop!(
             NFTPallet::create_collection(
-                Origin::signed(ALICE),
+                RuntimeOrigin::signed(ALICE),
```
[the remark this answers is not in the document]
Comment (Roznovjak) [reply nft/src/tests.rs:78]: IMHO to differentiate outer enums that are created by `construct_runtime` macro from nested enums from pallets.
You can see it in this example:
`RuntimeCall::Balances(pallet_balances::Call::transfer { dest: ALICE, value: 10 })`

File: test-utils/src/rpc_service.rs
Comment (Roznovjak) [reply test-utils/src/rpc_service.rs:1]: No. Deleted.

**Pull request merged**